app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1932,5 +1932,169 @@ class PatientRepositoryAndroidTest {
+
+    fun recordPatientAtFacility(
+        patientUuid: UUID,
+        patientName: String,
+        status: PatientStatus,
+        facility: Facility
+    ): PatientProfile {
+      val patientProfile = testData.patientProfile(patientUuid = patientUuid).let { patientProfile ->
+        patientProfile.copy(patient = patientProfile.patient.copy(status = status, fullName = patientName))
+      }
+      val bp = testData.bloodPressureMeasurement(patientUuid = patientUuid, facilityUuid = facility.uuid)
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+      bloodPressureRepository.save(listOf(bp)).blockingAwait()
+
+      return patientProfile
+    }
+
+    fun patientUuidsInAFacility(facility: Facility): List<UUID> {
+      return patientRepository.allPatientsInFacility(facility).blockingFirst().map { it.uuid }
+    }
+
+    // given
+    val (facilityA, facilityB, facilityC) = facilityRepository.facilities().blockingFirst()
+
+    val patientsInFacilityA = listOf(
+        recordPatientAtFacility(
+            UUID.fromString("e2a49529-3b46-4c9e-a7bc-af3090f1ecbb"),
+            "Patient C",
+            PatientStatus.ACTIVE,
+            facilityA
+        ),
+        recordPatientAtFacility(
+            UUID.fromString("7925e13f-3b04-46b0-b685-7005ebb1b6fd"),
+            "Patient A",
+            PatientStatus.ACTIVE,
+            facilityA
+        ),
+        recordPatientAtFacility(
+            UUID.fromString("ff1a267f-d2b1-4ca7-bc9d-1e035309c7cb"),
+            "Patient B",
+            PatientStatus.DEAD,
+            facilityA
+        )
+    )
+
+    val patientsInFacilityB = listOf(
+        recordPatientAtFacility(
+            UUID.fromString("a9ab09fd-fbee-4510-8e50-fd74f9a0f34b"),
+            "Patient E",
+            PatientStatus.DEAD,
+            facilityB
+        ),
+        recordPatientAtFacility(
+            UUID.fromString("51d7785c-dfe1-4018-8207-e74cd2377f5d"),
+            "Patient F",
+            PatientStatus.DEAD,
+            facilityB
+        ),
+        recordPatientAtFacility(
+            UUID.fromString("072df9ac-6a00-462f-b87e-1a9ca3b24c1b"),
+            "Patient D",
+            PatientStatus.ACTIVE,
+            facilityB
+        )
+    )
+
+    // when
+    val allPatientsInFacilityA = patientUuidsInAFacility(facilityA)
+    val allPatientsInFacilityB = patientUuidsInAFacility(facilityB)
+    val allPatientsInFacilityC = patientUuidsInAFacility(facilityC)
+
+    // then
+    assertThat(allPatientsInFacilityA)
+        .containsExactly(patientsInFacilityA[1].patient.uuid, patientsInFacilityA[0].patient.uuid)
+        .inOrder()
+
+    assertThat(allPatientsInFacilityB)
+        .containsExactly(patientsInFacilityB[2].patient.uuid)
+        .inOrder()
+
+    assertThat(allPatientsInFacilityC).isEmpty()
+  }
+
+  @Test
+  fun querying_all_patients_in_a_facility_with_bps_in_different_facilities_should_set_the_last_bp_to_the_latest_one() {
```

Comment:
Rephrase this to `querying_all_patients_in_a_facility_should_set_the_last_bp_to_the_last_recorded_one_within_any_facility`